### PR TITLE
FEATURE: theme settings in RAW hbs templates

### PR DESCRIPTION
**Pull request by OsamaSayegh**
https://meta.discourse.org/t/add-support-for-theme-settings/78103/21?u=osama

### CODE REVIEW

File: app/assets/javascripts/discourse/helpers/theme-setting-injector.es6 [hunk 1]
```diff
@@ -9,12 +9,28 @@ function inject(context, key, value) {
...
+
   if (!context.get("themeSettings")) {
     context.set("themeSettings", {});
   }
   context.set(`themeSettings.${key}`, value);
 }
 
+function injectPlainObject(context, key, value) {
+  if (!context.themeSettings) {
+    Object.assign(context, { themeSettings: {} });
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/helpers/theme-setting-injector.es6]: IE11 has not support of Object.assign AFAIK

File: app/assets/javascripts/discourse/helpers/theme-setting-injector.es6 [hunk 2]
```diff
@@ -9,12 +9,28 @@ function inject(context, key, value) {
...
+  }
+  Object.assign(context.themeSettings, { [key]: value });
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/helpers/theme-setting-injector.es6]: IE11 has not support of Object.assign AFAIK

**Comment by OsamaSayegh** [comment]
@jjaffeux thanks for pointing that out. I've switched to using underscore `assign` which I've verified works with IE 11.

**Comment by SamSaffron** [comment]
OK, lets try this out @vinothkannans  can you confirm it is all working for you now! 

Thanks Osama

**Pull request merged**